modules/core/src/main/scala/com/azavea/stac4s/StacCatalog.scala

```diff
@@ -58,7 +58,7 @@ object StacCatalog {
       ) =>
         StacCatalog.apply(
           version,
-          extensions,
+          extensions getOrElse Nil,
```

Comment:
This limits the set of changes here to only the encoders/decoders - which is nice because it reduces the downstream dependencies. But one thing that I _think_ happens as a result is if `stac4s` reads a catalog without `stac_extensions` it will always _write_ the same object with an empty list.

Should we change the encoder to behave the same as the decoder so it's consistent?

Reply:
_I_ don't think so, but I also don't think that fields should be both optional and possibly empty lists :man_shrugging:.

I can't think of a way that encoding an empty list will hurt anything. It might be annoying if you keep firing PATCHes at Franklin and despite your best efforts you just can't get rid of the `stac_extensions` field. I don't know.

Reply:
yeah I agree - I wish it was just an empty list too. The approach here makes sense to me.